### ISSUE TITLE
Does not allow definition of positional arguments from *args parent method

### ISSUE THREAD

**Issue by ioangatop** [microsoft/pyright#7095]
**Describe the bug**
When a parent method is defined with `*args` argument, it does not allow for a child method to define a specified positional variable due to _Positional parameter count mismatch_. Since the parent method allows any argument, I believe that we should be able to define the positional arguments that we want to use in a child class (now it seems to work only for the keyword arguments).

**Code**
```python
from typing import Any

class Parent:
    def method(self, *args: Any, **kwargs: Any) -> None:
        return

class ChildA(Parent):

    # No pyright error but runtime does not work with `var` as positional (expected)
    def method(self, *args: Any, var: int, **kwargs: Any) -> None:
        return
    
class ChildB(Parent):

    # Python runtime works as positional but pyright raises an error:
    # >>> ERROR Positional parameter count mismatch; base method has 3,
    # >>> but override has 4 (reportIncompatibleMethodOverride)
    def method(self, var: int, *args: Any, **kwargs: Any) -> None:
        return

ChildA().method(1)      # not working (expected) - correct to raise an error
ChildA().method(var=1)  # working (expected)
ChildB().method(1)      # working (expected) - incorrect to raise an error in the method definition
ChildB().method(var=1)  # working (expected)
```

Thank you for your time and your awesome tool!

**Comment by erictraut** [comment]
Thanks for the bug reoport. I agree this is incorrect behavior.

**Comment by erictraut** [comment]
Looking at this more closely, I think pyright's current behavior is actually correct. The `ChildA` class allows you to pass any number of positional arguments plus a keyword argument named `var`. That means this call works fine: `ChildA().method(1, var=1)`. If you pass the same arguments to `ChildB.method`, you will receive a runtime error. That means the override method is not compatible with the method it is overriding.
